library/ns1_zone.py

```diff
@@ -384,6 +436,73 @@ def compare_params(self, have, want):
                 diff[param] = wanted_val
         return diff
 
+    def get_changed_params(self, have, want):
+        """Gets Ansible params in want that have changed from have
+
+        :param have: Existing set of parameters
+        :type have: dict
+        :param want: Desired end state of parameters
+        :type want: dict
+        :return: Parameters in want that differ from have
+        :rtype: dict
+        """
+        diff = self.diff_params(have, want)
+        # perform deep comparison of secondaries if primary exists and has diff
+        if "primary" in have and "primary" in diff:
+            have_secondaries = have["primary"].get("secondaries")
+            want_secondaries = want["primary"].get("secondaries")
+            # if no difference in values, remove secondaries from diff results
+            if not self.diff_in_secondaries(
+                have_secondaries, want_secondaries
+            ):
+                diff["primary"].pop("secondaries", None)
+                # if secondaries was only key in primary, remove primary
+                if not diff["primary"]:
+                    diff.pop("primary", None)
+        return diff
+
+    def diff_in_secondaries(self, have_secondaries, want_secondaries):
+        """Performs deep comparison of two lists of secondaries, ignoring order.
+
+        :param have_secondaries: Existing secondaries list
+        :type have_secondaries: list
+        :param want_secondaries: Desired end state of secondaries list
+        :type want_secondaries: list
+        :return: Whether or not there is a difference between the lists
+        :rtype: bool
+        """
+        if want_secondaries is None:
+            # if no secondaries provided in params, no change
+            return False
+        elif have_secondaries is None:
+            # if no secondaries were already set, will be a change
+            return True
+        elif len(want_secondaries) != len(have_secondaries):
+            return True
+
+        have_hash = self.convert_secondaries_to_hash(have_secondaries)
+        want_hash = self.convert_secondaries_to_hash(want_secondaries)
+        diff = self.diff_params(have_hash, want_hash)
+        if diff:
+            return True
+
+        return False
+
+    def convert_secondaries_to_hash(self, secondaries):
```

Comment:
Not a big deal, but naming this just `hash` is a bit confusing to me, just from reading the function name I was thinking this was going to actually hash the entire thing and do the comparison off that, maybe rename this to hashmap?

Reply:
Oh, fair.  Maybe `convert_secondaries_to_dict()` would be ideal to keep the datastructure names pythonic?